packages/bpk-component-graphic-promotion/src/BpkGraphicPromo.tsx

```diff
@@ -40,64 +41,66 @@ export const TEXT_ALIGN = {
   start: 'start',
   center: 'center',
   end: 'end',
-};
+} as const;
 
 export type Props = {
-  className: ?string,
-  contentId: ?string,
-  tagline: ?string,
-  headline: string,
-  subheading: ?string,
-  sponsor: ?{
-    label: string,
-    logo: string,
-    altText: string,
-  },
-  buttonText: string,
-  onClick: () => void,
-  invertVertically: boolean,
-  textAlign: TEXT_ALIGN,
-  style: ?{},
+  className?: string | null;
+  contentId?: string | null;
+  tagline?: string | null;
+  headline: string;
+  subheading?: string | null;
+  sponsor?: {
+    label: string;
+    logo: string;
+    altText: string;
+  } | null;
+  buttonText: string;
+  onClick: () => void;
+  invertVertically?: boolean;
+  textAlign: typeof TEXT_ALIGN[keyof typeof TEXT_ALIGN];
+  style?: {};
 };
 
-const constructAriaLabel = ({
-  buttonText,
-  headline,
-  sponsor,
-  subheading,
-  tagline,
-}: Props) => {
-  const text = [];
-  const addText = (value) => value && text.push(value);
+const constructAriaLabel = (
+  buttonText: Props['buttonText'],
+  headline: Props['headline'],
+  sponsor: Props['sponsor'],
+  subheading: Props['subheading'],
+  tagline: Props['tagline'],
```

Comment:
Can we do:

```suggestion
  { buttonText, headline, sponsor, subheading, tagline }: Pick<Props, 'buttonText', 'headline', 'sponsor', 'subheading', 'tagline'>
```

Then. we could pass the whole `props` object rather than having to unbox it too. 🙂

Reply:
I'd been going with the simplest Flow->TS switch for readability, but now we've gone through the implementation and made sure it is the same output we can tweak types to be more TS-like.

`Pick` is a good suggestion:

https://github.com/Skyscanner/backpack/pull/2535/commits/985717eff2579de0ade20506d4aad2eb065d40e6

Unboxing props makes for more readable-at-a-glance code inside of GitHub, so following the convention of destructuring in the function component args here and pass them individually, since we're only going one level deep.